File: ros2_ws/src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class NumberPublisher : public rclcpp::Node
{
public:
    NumberPublisher() : Node("number_publisher"), number_(2)
    {
        number_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        number_timer_ = this->create_wall_timer(
            std::chrono::milliseconds(500),
            std::bind(&NumberPublisher::publish_number, this));

        RCLCPP_INFO(this->get_logger(), "Number publisher node has been started.");
    }

private:
    void publish_number()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        number_publisher_->publish(msg);
    }

    int64_t number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
    rclcpp::TimerBase::SharedPtr number_timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/number_publisher.py
import rclpy
from example_interfaces.msg import Int64
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus


class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.declare_parameter("number", 2)
        self.declare_parameter("publish_period", 1.0)

        self.number_ = self.get_parameter("number").value
        self.publish_period_ = self.get_parameter("publish_period").value

        # self.number_publisher_ = self.create_publisher(
        #     HardwareStatus, "HardwareStatus", 10
        # )
        self.number_publisher_ = self.create_publisher(Int64, "number", 10)
        self.number_timer_ = self.create_timer(self.publish_period_, self.my_publish_number)
        self.get_logger().info("Number publisher node has been started.")

    def publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.number_publisher_.publish(msg)

    def my_publish_number(self):
        msg = Int64()
        msg.data = self.number_
        # msg.temperature = 25.0
        # msg.are_motors_ready = True
        self.number_publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
